clients/python/moondream/cloud_vl.py

```
@@ -0,0 +1,161 @@
+        except Exception as e:
+            raise ValueError(f"Failed to convert image to JPEG.") from e
+
+    def _stream_response(self, req):
+        """Helper function to stream response chunks from the API."""
+        with urllib.request.urlopen(req) as response:
+            for line in response:
+                if not line:
+                    continue
+                line = line.decode("utf-8")
+                if line.startswith("data: "):
+                    try:
+                        data = json.loads(line[6:])
+                        if "chunk" in data:
+                            yield data["chunk"]
+                        if data.get("completed"):
+                            break
+                    except json.JSONDecodeError as e:
+                        raise ValueError(
+                            f"Failed to parse JSON response from server."
+                        ) from e
+
+    def caption(
+        self,
+        image: Union[Image.Image, EncodedImage],
+        length: str = "normal",
+        stream: bool = False,
+        settings: Optional[SamplingSettings] = None,
+    ) -> CaptionOutput:
+        encoded_image = self.encode_image(image)
+
+        payload = {
+            "image_url": encoded_image.image_url,
+            "length": length,
+            "stream": stream,
+        }
+
+        data = json.dumps(payload).encode("utf-8")
+        headers = {"X-Moondream-Auth": self.api_key, "Content-Type": "application/json"}
+        req = urllib.request.Request(
+            f"{self.api_url}/caption",
+            data=data,
+            headers=headers,
+        )
+
+        def generator():
+            for chunk in self._stream_response(req):
+                yield chunk
+
+        if stream:
+            return {"caption": generator()}
+
+        with urllib.request.urlopen(req) as response:
+            result = json.loads(response.read().decode("utf-8"))
+            return {"caption": result.get("caption", "")}
```

throw exception instead of defaulting to ""